core/src/test/java/io/aiven/kafka/tieredstorage/RemoteStorageManagerTest.java

```diff
@@ -0,0 +1,254 @@
+            arguments(null, ClosedByInterruptException.class),
+            arguments(RuntimeException.class, InterruptedException.class),
+            arguments(RuntimeException.class, ClosedByInterruptException.class),
+            arguments(StorageBackendException.class, InterruptedException.class),
+            arguments(StorageBackendException.class, ClosedByInterruptException.class)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideNonInterruptionExceptions")
+    void fetchSegmentNonInterruptionExceptionWhenGettingManifest(
+        final Class<Exception> outerExceptionClass,
+        final Class<Exception> exceptionClass
+    ) throws Exception {
+        final SegmentManifestProvider segmentManifestProvider = mock(SegmentManifestProvider.class);
+        when(segmentManifestProvider.get(any())).thenAnswer(invocation -> {
+            Exception innerException;
+            try {
+                innerException = exceptionClass.getDeclaredConstructor().newInstance();
+            } catch (final NoSuchMethodException e) {
+                innerException = exceptionClass.getDeclaredConstructor(String.class).newInstance("");
+            }
+
+            if (outerExceptionClass != null) {
+                throw outerExceptionClass.getDeclaredConstructor(String.class, Throwable.class)
+                    .newInstance("", innerException);
+            } else {
+                throw innerException;
+            }
+        });
+
+        final var config = Map.of(
+            "chunk.size", "1",
+            "storage.backend.class", "io.aiven.kafka.tieredstorage.storage.filesystem.FileSystemStorage",
+            "storage.root", targetDir.toString()
+        );
+        rsm.configure(config);
+        rsm.setSegmentManifestProvider(segmentManifestProvider);
+
+        assertThatThrownBy(() -> rsm.fetchLogSegment(REMOTE_LOG_METADATA, 0))
+            .isInstanceOf(RemoteStorageException.class)
+            .hasRootCauseInstanceOf(exceptionClass);
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideNonInterruptionExceptions")
+    void fetchSegmentNonInterruptionExceptionWhenGettingSegment(
+        final Class<Exception> outerExceptionClass,
+        final Class<Exception> exceptionClass
+    ) throws Exception {
+        // Ensure the manifest exists.
+        final ObjectKeyFactory objectKeyFactory = new ObjectKeyFactory("", false);
+        writeManifest(objectKeyFactory);
+
+        final ChunkManager chunkManager = mock(ChunkManager.class);
+        when(chunkManager.getChunk(any(), any(), anyInt())).thenAnswer(invocation -> {
+            Exception innerException;
+            try {
+                innerException = exceptionClass.getDeclaredConstructor().newInstance();
+            } catch (final NoSuchMethodException e) {
+                innerException = exceptionClass.getDeclaredConstructor(String.class).newInstance("");
+            }
+
+            if (outerExceptionClass != null) {
```

Comment:
I believe with current parameters the `outerExceptionClass` is never null, are there missing parameters or this if is unnecessary?

Reply:
I made it possible to be `null`